### ISSUE TITLE
Update issue on Windows

### ISSUE THREAD

**Issue by vivek1986** [VSCodium/vscodium#1010]
As I mentioned in the Issue-title(I-title), as VSCodium is a copy of VSCode minus the Telemetry, the VSCodium developers are unarguably and obviously responsible for making it exactly like VSCode in respect to all it's features which obviously includes the feature of Upgrades.

Right now the Upgrade features are really crappy(Windows 10 64Bit), if you can see it in the screengrab attached. Now let's see what happened when I clicked on "Download Update" as shown in screengrab, it downloaded the whole VSCodium zip source file, as in it wants to insinuate that it's User's responsibility to update the software manually. Now the very idea is absoutely unacceptable and absurd for starters.

**Again, it;s not User's responsibility to update the VSCodium manually at all... It must work exactly as the VSCode update feature does, i.e. download the update (most professionally the compressed Upgrade source only) and install the updates and remove the temporary downloaded compressed update source altogether in order to cleanup the User's important disk-space.**

Hope after providing the very valauable above INSTRUCTIONS FOR UPGRADE the VSCodium team will learn and make the Update process absolutely like that as their highest priority....
![image](https://user-images.githubusercontent.com/6047360/157447541-e49c4ade-1c13-4ce0-a244-27697398be28.png)

**Comment by daiyam** [comment]
Hi,

First of all, I will direct you to https://github.com/VSCodium/vscodium/blob/master/CODE_OF_CONDUCT.md#our-standards
Secondly, you don't give any details (how VSCodium was initially installed, which is your current version, ...)
Finally, you are free to provide any PR to fix any issue you are having.

PS: It's a very very small team which is maintaining this project.

**Comment by stripedpajamas** [comment]
I will add that the update process in general works exactly like VSCode's. So your experience is more indicative of a bug. Please provide more details as @daiyam has requested.

**Comment by mattkwarren** [comment]
"the VSCodium developers are unarguably and obviously responsible for making it exactly like VSCode in respect to all it's features which obviously includes the feature of Upgrades"

They aren't responsible for anything. I didn't pay for VS Codium, nor do I have any agreement with them. VS Codium is based on the source code of VS Code. There is no requirement for anyone to do anything with VS Codium, other than what is stated in the license agreement. As far as I know, they are following all agreements by making the source code freely available.

"It must work exactly as the VSCode update feature does"

Says who? You already use Windows. If VS Codium isn't the exact clone of VS Code you want it to be, why not just use VS Code? You already use Windows, so I can't imagine you care about the lack of telemetry that much. Your whole OS is already reporting to Microsoft.

Bonus: You can contribute fixes if you think something is broken.

**Comment by hazarek** [comment]
> You already use Windows, so I can't imagine you care about the lack of telemetry that much. Your whole OS is already reporting to Microsoft.

🤣

**Comment by ocdkerosine** [comment]
> You already use Windows, so I can't imagine you care about the lack of telemetry that much. Your whole OS is already reporting to Microsoft.

xD hands down the coldest thing i've read on github. @vivek1986 he's right too. you should switch to ParrotOS and feel the real pain of living a life without telemetry. using vscodium on windows is like trying to stop a nuke with your bare hands.

**Comment by argv-minus-one** [comment]
Linux user here. So, @vivek1986 was all kinds of rude, and I very much appreciate that the VSCodium developers have gone to the trouble of making a malware-free VSCode…but I can't help agreeing that it would be nice if VSCodium could update itself, at least on non-Microsoft platforms.

I'm guessing Microsoft VSCode's self-update code is proprietary?

**Comment by stripedpajamas** [comment]
@argv-minus-one we have done quite a bit of work to get auto-updating working smoothly for VSCodium. 

The main piece is an API with the same interface as Microsoft's update API running at VSCodium.now.sh. This gets put into the product.json at build time. The VSCode codebase uses this URL to check for updates, and to download the latest version if it finds it's behind. 

I use a Mac so I know Mac auto-updating works. I was pretty sure Windows auto-updating was working as well, but there are a bunch of different ways of installing VSCode/VSCodium on Windows (system setup, user setup, zip, for both arch's), so maybe only some of them have auto-updating working. 

As far as I remember, the codebase itself doesn't do the same procedure when on Linux. It still checks for updates, but the returned URL is just the tar.gz file, which would require manual extraction etc. This being said, Linux users are supposed to rely on their package managers to handle the update. 

I haven't run VSCode or VSCodium on Linux in a while though...if the update experience of VSCode is better than VSCodium on your OS I'd love some more details so that perhaps we can get back to update parity.

**Comment by argv-minus-one** [comment]
>if the update experience of VSCode is better than VSCodium on your OS I'd love some more details so that perhaps we can get back to update parity.

I've never used Microsoft VSCode on Linux, now that you mention it. I was thinking of the update experience on Windows, but Microsoft VSCode's update experience on Linux could be exactly the same as VSCodium for all I know.

>Linux users are supposed to rely on their package managers to handle the update.

That explains it! I installed VSCodium by hand from the tar archive. (I don't like installing third-party packages with the system package manager because that allows them to run arbitrary scripts as root.) It makes sense that it can't automatically update itself when it was manually installed.

**Comment by vivek1986** [comment]
@hazarek @ocdkerosine @mattkwarren @argv-minus-one @stripedpajamas, basically, everyone who's false criticising me for going out of my way and pointing out that the VSCodium Upgrade process **sucks a lot which is unacceptable**, by arguments not making sense or calling me rude, if you all had read the message thoroughly then you would have noted and accepted unargumentably that my message is nothing but CONSTRUCTIVE CRITICISM, which is must, will always be there and is absolutely necessary for pushing (sometimes harder than ever) the Software Developers(and any IT guy) who are responsible and accountable for making the Software time-saving, easy-to-operate and surpassing the expectations of User in each and every aspect and step. No one(and obviously not me) is asking for Rocket science here, JUST DOWNLOAD AND TAKE A LOOK AT ANY MEDIUM SIZED EDITOR TODAY(FORGET VSCODE, LET'S TAKE ATOM EDITOR, JETBRAINS PHPSTORM, DREAMWEAVER, SLICKEDIT ETC.), THEY DEFINITELY PROVIDE FULL-FLEDGE PLUGIN INSTALLATION, CODE RUNNING AND COMPILING FACILITIES AND ALSO FULL-FLEDGE UPGRADE SYSTEM WITHOUT FAIL....

I don't get it what is it so hard and unachievable that the same developers who made the VScodium without telemetry can not clone the upgrade feature such that it downloads, installs the update and then deletes whatever file it has downloaded.

HELL, EVEN IF I AM A WEB DEVELOPER, I TOO KNOW BATCH/CMD/PWOERSHELL HYBRID-SCRIPTING ENOUGH TO MAKE A SCRIPT THAT WOULD CLOSE THE RUNNING INSTANCES OF VSCODIUM, DOWNLOAD THE UPDATE PACKAGE, THEN EXTRACT THE ZIP FILE AND REPLACE THE CONTENTS OF VSCODUM PROGRAM FILES WITH THE CONTENTS OF EXTRACTED FOLDER, THEN BOOM !!! VSCODIUM IS NOW UPGRADED WITHOUT USER HAVING TO MANUALLY DOWNLOAD THE FILE AND EXTRACTING AND REPLACING. See this script, it might not be fine-grained, but improvement and functionality on this lines of this if integrated in the VSCodium setup itself and runs whenever new version is available would be really an improvement of next level:
```bat
@echo off && setlocal
@pushd %~dp0 & fltmc | find "." && (powershell start '%~f0' ' %*' -verb runas 2>nul && exit /b)
taskkill /f /t /im vscodium* && taskkill /f /t /im vscodium*
set "DLDir=%USERPROFILE%\Downloads" && del /f /s /q "%DLDir%\VSCodium*.zip"
if exist "%DLDir%\VSCodiumUpdate" rd /s /q "%DLDir%\VSCodiumUpdate"
powershell "$cli = New-Object System.Net.WebClient;$cli.Headers['User-Agent'] = 'WillisBrowser';$cli.DownloadFile('https://github.com/VSCodium/vscodium/releases/download/1.65.2/VSCodium-win32-x64-1.65.2.zip', '%DLDir%\VSCodium-win32-x64-1.65.2.zip');"
powershell Expand-Archive "%DLDir%\VSCodium*.zip" -DestinationPath "%DLDir%\VSCodiumUpdate"
xcopy /s/e /v /h /k /x /y "%DLDir%\VSCodiumUpdate" "%PROGRAMFILES%\VSCodium"
if exist "%DLDir%\VSCodiumUpdate" rd /s /q "%DLDir%\VSCodiumUpdate"
endlocal && exit
```

@daiyam You very quickly asked that, how VSCodium was installed and what's the version, well, let me tell you: I had installed VSCodium with Windows executable format file, and with recent versions maybe after 1.64.x(this is an assumption, I don't remember the exact version number) the upgrade process is not working and also let me point out that it doesn't matter VScodium is either User setup or System setup, it's upgrade process must be streamlined enough to not let User intervene when new versions are released and upgrade is due...

Don't know what more info I can provide to make you figure out the issue or problem area for Upgrades but if I re-iterate what I said earlier, see in the screenshot attached in Question, when I click the "Download Update" link, it just downloads the newest version zipfile and then poof, nothing in the direction of upgrade happens, if I close VSCodium and restart the VSCodium is still un-upgraded and again there shows notification "There is an available update >> Download Update or >> Later or >> Releae Notes buttons" and in my ~/Downloads(Windows ofcourse, i.e. %USERPROFILE%\Downloads) folder there is a newly downloaded zipfile present, but unused and unextracted...

**Comment by daiyam** [comment]
Whatever the issue you are having, your initial and second posts are not constructive at all, are lacking any kind of useful informations and don't show any empathy.

You could have posted a more simple message as:

> Hi,
>
> I'm on Windows 10 64bits and the update process doesn't work. 

Then, probably, we would have spent some of <ins>**our time**</ins> to figure out your issue.

**Comment by vivek1986** [comment]
@daiyam You definitely have tendency to intentionally focus on just the somewhat harsh words and will-fully ignoring the whatever possible useful first-hand info that I have already provided in both my posts, expecting, a level of understanding, maturity and non-complancence from you. YOU SERIOUSLY SAYING THAT YOU DIDN'T PROPERLY READ AND/OR GOT ZERO IDEA AFTER SEEING THE SCREENSHOT AND READING THIS "..it just downloads the newest version zipfile and then poof, nothing in the direction of upgrade happens, if I close VSCodium and restart the VSCodium is still un-upgraded and again there shows notification "There is an available update >> Download Update or >> Later or >> Releae Notes buttons" and in my ~/Downloads(Windows ofcourse, i.e. %USERPROFILE%\Downloads) folder there is a newly downloaded zipfile present, but unused and unextracted..." .

IF SO AS USER MY RESPONSIBILITY IS WAY MORE THAN OVER AND I'VE BEEN A GOOD SAMARITAN IN POINTING OUT WHATEVER I EXPERIENCED WRONG THAT'S GOING ON WITH VSCODIUM(AND CAN'T BE EXPECTED TO PROVIDE MORE THAN WHAT I SAW & EXPERIENCED AND ALREADY PROVIDED)(I HAVEN'T TESTED THIS ON LINUX AS I DON'T HAVE LINUX) AND CAN'T PROVIDE MORE INFO, UNLESS I HAVE TO DRAG MYSELF DOWN INTO VSCODIUM DEVELOPEMENT UNWILLINGLY JUST TO HELP YOU ANALYSE AND UNDERSTAND ISSUE REQUIREMENT FOR STARTERS... NO WAY THIS LEVEL OF USER-INTERACTON AND USER-ASSISTANCE WAS EXPECTED FROM YOU, WELL LASTLY I(BEING TIRED OF THIS CHARADE) WOULD LEAVE IT ON YOU, IF YOU REALLY ARE WILLING, FROM THE DEEP DOWN OF YOUR SOUL AND BETTER ANGELS, TO FIX THE UPGRADE PROCESS THEN READ MY EARLIER MESSAGES AS MANY TIMES AS YOU NEED TO DRILL DOWN AND GATHER THE IDEA OF WHAT'S WRONG WITH UPGRADE(ALSO TAKE LOOK AT THE BATCH SCRIPT I PROVIDED OR HAVE ANY BATCH/POWERSHELL EXPERT TAKE LOOK IF THAT HELPS) AND TRY TO FIX IT, CAUSE I AM DONE EXPLAINING THE SAME THING OVER AND OVER AGAIN AND NOT GONNA SUGARCOAT(AND NOT GONNA SHOW ANYMORE POLITENESS/EMPATHY) ANYMORE THAN I ALREADY HAVE, JUST BECAUSE YOU CANNOT WORK UNDER SOMEWHAT PRESSURE AND CRITICISM...
ANY OTHER DEVELOPER TEAM WOULD'VE DEFINITELY TAKEN THIS AS HEALTHY CUSTOMER ISSUE & PROFESSIONAL CHALLENGE AND MAYBE WOULD'VE FIXED IT BY NOW, BUT YOU KNOW THAT'S JUST ALL OF THEM, RIGHT ?

**Comment by hazarek** [comment]
> ANY OTHER DEVELOPER TEAM WOULD'VE DEFINITELY TAKEN THIS AS HEALTHY CUSTOMER ISSUE & PROFESSIONAL CHALLENGE AND MAYBE WOULD'VE FIXED IT BY NOW, BUT YOU KNOW THAT'S JUST ALL OF THEM, RIGHT ?

VScodium usere are not customers. VScode has over 5k issues on Github. A bug I've been tracking hasn't been fixed for 4 years. Although there are Professional commercial Microsoft developers behind it.

**Comment by vivek1986** [comment]
@hazarek Not a matured and acceptable "response" or excuse at all. Just because someone(regardless whether they're popular and king-size firms/companies or not) are not able to do it(maybe due to other priorities or not have the resources and abilities to fix it) doesn't mean we(or you) should be "carbon-copy" of them or assumingly take inspiration from them. And still to answer your concern, the Upgrade of software comes under a very extemely fundamental feature, if that feature is tiresome and manual like it is currently, then any User/Customer(I would call them Customers only rightfully) can expect what from future versions ? You feel me ?

**Comment by daiyam** [comment]
This is going out of hand. @vivek1986, your behaviour isn't acceptable.

**Comment by daiyam** [comment]
I've verified the update process on Windows, here my findings:

- `VSCodium.msi`: download zip
- `VSCodium-updates-disabled.msi`: no update menu
- `VSCodium.zip`: download zip
- `VSCodiumSetup.exe`: install update
- `VSCodiumUserSetup.exe`: install update

The only issue is with `VSCodium.msi` which is downloading the zip instead of performing the update.

**Comment by daiyam** [comment]
Hi, I've unlocked the conversation but I expect everybody to focus on the issue.

**Comment by daiyam** [comment]
The update service `/src/vs/platform/update/electron-main/updateService.win32.ts`and update server (https://github.com/VSCodium/update-api) will have to be updated to support MSI installs.

The service is defaulting to an archive update since it doesn't detect the MSI install.

**Comment by daiyam** [comment]
It's fixed since `1.72.0`
